Shopify/shopify_app
# Support Rails 7

Closes https://github.com/Shopify/shopify_app/issues/1339

### What this PR does

- Adds support to work with Rails 7 projects
- Updates the install generator to work with Importmap instead of Webpack
- Fix the regex to properly whitelist Ngrok tunnel hostnames

### Reviewer's guide to testing

1. Ensure you have Rails 7: `rails -v`
2. `rails new my_app`
3. `bundle add shopify_app --git 'git://github.com/Shopify/shopify_app.git' --branch 'rails-7'`
4. `rails generate shopify_app`
5. `rails db:migrate`
6. `ngrok http 3000` and copy the HTTPS URL
7. Create a new app in [Partners](https://partners.shopify.com/), adding the Ngrok URL followed by `/auth/shopify/callback` in the `Allowed redirection URLs` field
8. Copy the generated api key and secret
9. `SHOPIFY_API_KEY=x SHOPIFY_API_SECRET=y rails server`
10. Go to the Ngrok URL, install the app on a store and check that it works

### Checklist

Before submitting the PR, please consider if any of the following are needed:
